pkg/ctrl/watch/manager.go

```diff
@@ -0,0 +1,117 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	log = ctrl.Log.WithName("watch")
+)
+
+// This controller will be moved to RuntimeReconcilerInterface later
+type Controller interface {
+	// Reconciler performs a full reconciliation for the object referred to by the Request.
+	// The Controller will requeue the Request to be processed again if an error is non-nil or
+	// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+	Reconcile(context context.Context, req ctrl.Request) (ctrl.Result, error)
+
+	// ControllerName gets the name of controller
+	ControllerName() string
+
+	// ManagedResource which is handled by controller
+	ManagedResource() client.Object
+}
+
+func SetupWatcherWithReconciler(mgr ctrl.Manager, options controller.Options, r Controller) (err error) {
+	options.Reconciler = r
+	c, err := controller.New(r.ControllerName(), mgr, options)
+	if err != nil {
+		return err
+	}
+
+	runtimeEventHandler := &runtimeEventHandler{}
+	err = c.Watch(&source.Kind{Type: r.ManagedResource()}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: runtimeEventHandler.onCreateFunc(r),
+		UpdateFunc: runtimeEventHandler.onUpdateFunc(r),
+		DeleteFunc: runtimeEventHandler.onDeleteFunc(r),
+	})
+	if err != nil {
+		log.Error(err, "Failed to watch JindoRuntime")
+		return err
+	}
+
+	statefulsetEventHandler := &statefulsetEventHandler{}
+	err = c.Watch(&source.Kind{Type: &appsv1.StatefulSet{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    r.ManagedResource(),
+	}, predicate.Funcs{
+		CreateFunc: statefulsetEventHandler.onCreateFunc(r),
+		UpdateFunc: statefulsetEventHandler.onUpdateFunc(r),
+		DeleteFunc: statefulsetEventHandler.onDeleteFunc(r),
+	})
+	if err != nil {
+		return err
+	}
+
+	daemonsetEventHandler := &daemonsetEventHandler{}
+	err = c.Watch(&source.Kind{Type: &appsv1.DaemonSet{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    r.ManagedResource(),
+	}, predicate.Funcs{
+		CreateFunc: daemonsetEventHandler.onCreateFunc(r),
+		UpdateFunc: daemonsetEventHandler.onUpdateFunc(r),
+		DeleteFunc: daemonsetEventHandler.onDeleteFunc(r),
+	})
+	if err != nil {
+		return err
+	}
+
+	return
+}
+
+// isObjectInManaged checks if the object is managed by Fluid runtime controller
+func isObjectInManaged(obj metav1.Object, r Controller) (managed bool) {
+	if controllerRef := metav1.GetControllerOf(obj); controllerRef != nil && isOwnerMatched(controllerRef, r) {
+		log.V(1).Info("obj.onCreateFunc ControllerRef will handle the object due to  owner reference is matched with runtime", "name", obj.GetName(), "namespace", obj.GetNamespace())
```

Comment:
`isObjectInManaged()` is also used in `onUpdateFunc` and `onDeleteFunc`. I suggest to change the log content here. Or maybe log in corresponding funcs.